### DISCUSSION TITLE
SCReAM congestion control

### DISCUSSION THREAD

**Discussion by IngJohEricsson** [pion/webrtc#2101] [Ideas]
Hi
If you are interested.

SCReAM (Self-clocked Rate Adaptation for Multimedia) congestion control (RFC8298) is available as open source from https://github.com/EricssonResearch/scream 
The gstscream subfolder contains build scripts to create the sender and receiver congestion control parts of SCReAM 

/Ingemar

**Comment by mengelbart** [comment]
Hi @IngJohEricsson ! 

Thanks for the pointer! We have a SCReAM implementation using CGO, which is based on the implementation in the `code` subdirectory of the repository you linked, though it is using a [fork](https://github.com/mengelbart/scream-go) from some time ago. I think the Rust implementation did not yet exist when I forked it. The CGO integration in Pion is on a separate [branch in Pions interceptor repository](https://github.com/pion/interceptor/tree/feat/scream-cgo). Maybe we can also find a way to integrate the new Rust implementation in Pion. 

Eventually, we'd also like to have a pure Go implementation so everyone can use it without using CGO.

We also recently finished an implementation of GCC, and we plan to implement a test runner to automatically run the test cases from [RFC 8867](https://www.rfc-editor.org/rfc/rfc8867.html) for all of the bandwidth estimators in Pion!

**Comment by IngJohEricsson** [comment]
Hi

Awfully sorry for the slow response. Let me know if you encunter issues with the SCReAM implementation. Hope that the RTT will be less than 2.5 months then :-)
/Ingemar
